app/src/main/kotlin/com/adesso/movee/scene/search/SearchViewModel.kt

```diff
@@ -31,11 +31,12 @@ class SearchViewModel @Inject constructor(
         val query = text?.trim() ?: return
         if (query.length > MIN_SEARCHABLE_LENGTH) {
             multiSearchJob?.cancel()
-            multiSearchJob = bgScope.launch {
-                val searchResult = multiSearchUseCase.run(MultiSearchUseCase.Params(query))
-                onUIThread {
-                    searchResult.either(::handleSearchFailure, ::postMultiSearchResult)
-                }
+
+            multiSearchUseCase.invoke(
+                viewModelScope,
+                MultiSearchUseCase.Params(query)
+            ) {
+                it.either(::handleSearchFailure, ::postMultiSearchResult)
```

Comment:
postMultiSearchResult() uses LiveData's setValue() functions which must be called from main thread. Either we need to use postValue() on both LiveDatas we wan't their values to change, or switch context when calling postMultiSearchResult()